docs/errors.md

```diff
@@ -0,0 +1,67 @@
+## Error Reporting
+
+Since the 3scale toolbox is used from scripts / pipelines / etc, any error must be reported without any ambiguity.
+
+## Table of contents
+* [Unix Return Code](#unix-return-code)
+* [Error Report Fields](#error-report-fields)
+* [Error Types](#error-types)
+* [Error Code Table](#error-code-table)
+
+### Unix Return Code
+
+If the requested operation cannot be completed, the Unix return code is *1* and if the operation has been completed successfully the Unix return code is *0*.
+
+### Error Report Fields
+
+When an error occurs, the toolbox writes to *stderr* an structured error report with some fields.
+
+```
+{
+  "code": string,
+  "message": string,
+  "class": string,
+  "stacktrace": array of string
+}
+```
+
+| Field | Description |
+| --- | --- |
+| code | Error code. See [table](#error-code-table) |
+| message | Description text of the error |
+| class | Class of the error |
```

Comment:
Maybe specify that class is a Ruby class?
Before looking at the code I thought it could be a Ruby class or the "class" of the error (conceptually)

Reply:
IMO user does not need to know that.